Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py

```diff
@@ -334,7 +334,9 @@ def set(self, issue, assignee=None, opened=None, why=None, project=None, compone
             if not isinstance(assignee, User):
                 raise TypeError("Must assign to '{}', not '{}'".format(User, type(assignee)))
             issue._assignee = self.user(name=assignee.name, username=assignee.username, email=assignee.email)
-            update_dict['assignees'] = [issue._assignee.username]
+            assignees = [issue._assignee.username]
+            if self.add_assignees(issue, assignees) != assignees:
+                return None
```

Comment:
Not sure if the whole tool should fail if assignment isn't successful, previously the error was never caught.

Reply:
I had this same thought.

Ultimately, it's a pretty rare situation that would require someone making a single call that bot set the assignee and did something else. While that's possible (and relevant for bugzilla and radar, where such operations _are_ done in a single request), it's a rare enough situation that I'm not inclined to advocate increasing code complexity to deal with it.